resources/assets/js/components/VenueAddress.vue

```diff
@@ -0,0 +1,79 @@
+<template>
+  <div>
+    <div class="form-group">
+      <label :for="$id('address-autocomplete')">{{ __('events.field_event_venue') }}:</label>
+      <vue-google-autocomplete
+          :id="$id('address-autocomplete')"
+          name="location"
+          classname="form-control"
+          :placeholder="__('events.field_venue_placeholder')"
+          @placechanged="placeChanged"
+          aria-describedby="locationHelpBlock"
+          types="geocode"
+          ref="autocomplete"
+      >
+      </vue-google-autocomplete>
+      <p class="text-danger" v-if="error">{{ error }}</p>
+
+      <small id="locationHelpBlock" class="form-text text-muted">
+        {{ __('events.field_venue_helper') }}
+      </small>
+      <b-btn variant="link" size="sm" v-if="groupLocation" @click="useGroup" class="pl-0" :disabled="currentValue === groupLocation">
+        {{ __('events.field_venue_use_group') }}
+      </b-btn>
+    </div>
+  </div>
+</template>
+<script>
+import Vue from 'vue'
+import VueGoogleAutocomplete from 'vue-google-autocomplete'
+import UniqueId from 'vue-unique-id';
```

Comment:
What is vue-unique-id for?

Reply:
You need an id on an `input `in order to be able to reference it using `for `with a `label`.  

But you want all the ids on the page to be unique.  This can be a problem with Vue, because it becomes quite easy to end up with duplicate ids by adding multiple copies of a component.  

vue-unique-id helps solve this by allowing you to easily generate a globally unique id.